Add Aspire to examples

holytshirt opened the pull request.

holytshirt committed:
got create and migration working

holytshirt committed:
Implement new solution structure and update package versions

- Added a new solution file `Brighter.slnx` with multiple project configurations for various samples and tasks.
- Updated package versions in `Directory.Packages.props` to 9.2.1 for `Aspire.Hosting.AppHost` and `Aspire.Hosting.MySql`, and added new dependencies.
- Refactored connection string retrieval in `Extensions.cs` and `Program.cs` for better clarity and consistency.
- Updated project references in `GreetingsWeb.csproj` and `MigrationService.csproj` to ensure proper linking.
- Cleaned up unused package versions and improved project structure in `WebAPI_EFCore` samples.

holytshirt committed:
Refactor database entity and API endpoint configurations

- Updated `OutboxFactory.cs` to correct the order of parameters in the MySQL outbox creation method.
- Renamed the `Recipient_Id` column to `RecipientId` in the initial migration file for consistency.
- Modified HTTP test scripts to use a base URI variable for cleaner endpoint management.
- Changed the `DbSet` property name from `People` to `Person` in `GreetingsEntityGateway` for clarity.
- Updated various handler classes to reference the new `Person` property instead of `People`.
- Adjusted the greeting creation endpoint to include the person's name in the URL path for better RESTful design.

holytshirt committed:
Add RabbitMQ support to the project

- Added `Aspire.Hosting.RabbitMQ` package version 9.2.1 to `Directory.Packages.props`.
- Updated `Program.cs` to include RabbitMQ configuration and reference in the project setup.
- Modified project file `WebAPI_EFCore.AppHost.csproj` to include RabbitMQ as a package reference.

slang25 commented on Directory.Packages.props:
```diff
@@ -49,11 +54,14 @@
     <PackageVersion Include="Microsoft.Extensions.Hosting.Abstractions" Version="9.0.4" />
     <PackageVersion Include="Microsoft.Extensions.Http" Version="9.0.4" />
+    <PackageVeroisn Include="Microsoft.Extensions.Http.Resilience" Version="9.4.0" />
+    <PackageVersion Include="Microsoft.Extensions.Http.Resilience" Version="9.4.0" />
```
```suggestion
```
Looks like this line got doubled up

holytshirt committed:
refactor: database connection handling and update package dependencies

- Added `Aspire.Pomelo.EntityFrameworkCore.MySql` package version 9.3.0 to `Directory.Packages.props`.
- Refactored `ConnectionResolver.cs` to simplify database connection string retrieval using a switch expression.
- Updated `DbConnectionFactory.cs` to remove commented-out code and improve clarity.
- Enhanced `GreetingsDbFactory.cs` and `SalutationsDbFactory.cs` to configure database connections with new options for `SelectingProcessorAccessorOptions`.
- Modified transport configuration methods in `ConfigureTransport.cs` to accept connection strings for better flexibility.
- Updated various project files to ensure proper references and configurations for the new database setup.
- Adjusted HTTP test scripts to reflect changes in connection strings and endpoint management.

holytshirt committed:
chore: update Aspire package versions to 9.3.1 for consistency and compatibility

holytshirt committed:
feat: add SalutationsMigrator service and update test data for consistency
